dspace-spring-rest/src/test/java/org/dspace/app/rest/builder/WorkflowItemBuilder.java

```diff
@@ -101,22 +106,48 @@ private void deleteWsi(WorkspaceItem dso) throws Exception {
             WorkspaceItem attachedDso = c.reloadEntity(dso);
             if (attachedDso != null) {
                 workspaceItemService.deleteAll(c, attachedDso);
+                item = null;
             }
             c.complete();
         }
 
         indexingService.commit();
     }
 
+    private void deleteItem(Item dso) throws Exception {
+        try (Context c = new Context()) {
+            c.turnOffAuthorisationSystem();
+            Item attachedDso = c.reloadEntity(dso);
+            if (attachedDso != null) {
+                // if we still have a reference to an item it could be an approved workflow or a rejected one. In the
+                // last case we need to remove the "new" workspaceitem
+                WorkspaceItem wi = workspaceItemService.findByItem(c, item);
+                if (wi != null) {
+                    workspaceItemService.deleteAll(c, wi);
+                } else {
+                    itemService.delete(c, attachedDso);
+                }
+            }
+            c.complete();
+        }
+
+        indexingService.commit();
+    }
 
     @Override
     public void cleanup() throws Exception {
         if (workspaceItem != null) {
+            System.out.println("remove workspaceitem");
```

Comment:
We should remove this `System.out.println` ;)

Reply:
If we want to keep some notice of what the code is doing, maybe change these to log.debug().